Microsoft.DotNet.Try.Project/FSharpMethods.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.DotNet.Try.Protocol;
+using Buffer = Microsoft.DotNet.Try.Protocol.Buffer;
+
+namespace Microsoft.DotNet.Try.Project
+{
+    internal class FSharpMethods
+    {
+        private const string FSharpRegionStart = "//#region";
```

Comment:
>"//#region" [](start = 49, length = 11)

Is this ideal? Any suggestions for making it more natural in F#?

Reply:
Not really; F# doesn't have the equivalent of `#region` or any concept of a code block.  The closest thing we have is [computation expressions](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/computation-expressions), but that would require the writer of the documentation/sample to define a new builder for each span of code.  E.g., the backing code *could* look like this:

``` F#
type DemoRegionBuilder() =
    member __.Zero() = ()

// ... elsewhere
let demo_region_1 = new DemoRegionBuilder()
let demo_region_2 = new DemoRegionBuilder()
// .. repeat for every equivalent of `#region demo_region_X` that would be found in C#

[<EntryPoint>]
let main(args: string[]) =
    let _ =
        demo_region_1 {
            printfn "hello, world" // this line appears in the web page
        }
    0
```

With the markdown looking like this:

``` markdown
` ` ` fsharp --project ./MySampleProject.fsproj --source-file ./MySampleFile.fs --region demo_region_1
printfn "hello, world" // this line appears in the web page
` ` `
```

Adding @KevinRansom to see if he has any additional insight, but I'm still leaning towards specially-formatted comments, because that's essentially what C#/VB `#region`s are anyways.